### DISCUSSION TITLE
Error while running Cordova

### DISCUSSION THREAD

**Discussion by deicool** [apache/cordova#371] [General]
Hello

I am a newbie.

I installed cordova for windows. I was playing around a bit with the startup (https://cordova.apache.org/).

When I try to run cordova for android, i get the following error:

Built the following apk(s):
      
-----------------------------------------------------------------------------------------------------------------
 D:\Saksham\cordova\DaTaraMandal\platforms\android\app\build\outputs\apk\debug\app-debug.apk
Checking Java JDK and Android SDK versions
ANDROID_SDK_ROOT=C:\Users\Admin\AppData\Local\Android\sdk (recommended setting)
ANDROID_HOME=undefined (DEPRECATED)
Using Android SDK: C:\Users\Admin\AppData\Local\Android\sdk
Command failed with exit code 1: apkanalyzer manifest target-sdk D:\Saksham\cordova\DaTaraMandal\platforms\android\app\build\outputs\apk\debug\app-debug.apk
'sh' is not recognized as an internal or external command,
operable program or batch file.
-----------------------------------------------------------------------------------------------------------------

Any idea what to do?

**Comment by breautek** [comment]
Can you run `cordova requirements` and `cordova info` and paste the output.

**Comment by deicool** [comment]
*cordova requirements*
---------------------------------------------------------------------------------------------------------------
Requirements check results for android:
Java JDK: installed 1.8.0
Android SDK: installed true
Android target: installed
android-33,android-32,android-30,android-29,android-28,android-23
Gradle: installed D:\Program Files\Android\Android
Studio171\gradle\gradle-4.1\bin\gradle

Requirements check results for browser:

Requirements check results for ios:
Apple macOS: not installed
Error: Cordova tooling for iOS requires Apple macOS
Some of requirements check failed
---------------------------------------------------------------------------------------------------------------


*cordova info*
---------------------------------------------------------------------------------------------------------------
Cordova Packages:

    cli: 11.0.0
        common: 4.1.0
        create: 4.0.0
        lib: 11.0.0
            common: 4.1.0
            fetch: 3.1.0
            serve: 4.0.0

Project Installed Platforms:

    android: 10.1.2
    browser: 6.0.0
    ios: 6.2.0

Project Installed Plugins:


Environment:

    OS: Microsoft Windows 10 Pro 10.0.19045 (19045) (Windows 10.0.19045) x64
    Node: v18.12.1
    npm: 8.19.2

android Environment:

    android:
ERROR: Command failed with exit code 1: avdmanager list target
'avdmanager' is not recognized as an internal or external command,
operable program or batch file.


ios Environment:

    xcodebuild:
ERROR: Command failed with exit code 1: xcodebuild -version
'xcodebuild' is not recognized as an internal or external command,
operable program or batch file.


Project Setting Files:

    config.xml:
<?xml version='1.0' encoding='utf-8'?>
<widget id="io.cordova.hellocordova" version="1.0.0" xmlns="
http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
    <name>HelloCordova</name>
    <description>Sample Apache Cordova App</description>
    <author ***@***.***" href="https://cordova.apache.org
">
        Apache Cordova Team
    </author>
    <content src="index.html" />
    <allow-intent href="http://*/*" />
    <allow-intent href="https://*/*" />
</widget>

    package.json:
--- Start of Cordova JSON Snippet ---
{
  "platforms": [
    "browser",
    "android",
    "ios"
  ]
}
--- End of Cordova JSON Snippet ---

---------------------------------------------------------------------------------------------------------------

Deepak
"The greatness of a nation can be judged by the way its animals are treated
- Mahatma Gandhi"

+91 73500 12833
***@***.***

Facebook: https://www.facebook.com/deicool
LinkedIn: www.linkedin.com/in/deicool

"Plant a Tree, Go Green"

Make In India : http://www.makeinindia.com/home

**Reply by breautek** [reply]
Thank you.

There's quite a transition between cordova-android@10 and cordova-android@11 where the underling android SDK has several breaking changes and requires using a specific toolset that aren't backwards compatible. I suspect that you might be mixing some of these toolsets.

For cordova-android@10, double check the following things:

1. `JAVA_HOME` points to a JDK 8 installation (based on your response, it appears to be)
2. Try setting `ANDROID_HOME` in addition to the `ANDROID_SDK_ROOT`, they should be equal. Sometimes, tooling looks for `ANDROID_HOME`, other times `ANDROID_SDK_ROOT`. Google has flip-flopped on this, and `ANDROID_HOME` is no longer deprecated.
3. Ensure that Build tools 30.0.3 is installed and is in your `PATH`. I'm not familiar with windows variables, but should be located at `$ANDROID_HOME/build-tools/30.0.3`. If the folder is missing, you must install it using the Android SDK manager.
4. Additionally, the following should also be in your `PATH` system variable:
- `$ANDROID_HOME/emulator`
- `$ANDROID_HOME/platform-tools`
- `$ANDROID_HOME/tools/bin`
- `$ANDROID_HOME/tools`
5. If you aren't using a physical device, you may need to create an AVD (Android Virtual Device) manually before you can run `cordova run android`.

If you have Android Studio installed, it provides a GUI interface to both the SDK Manager and an AVD Manager.

Note that cordova-android@10 is no longer supported as Google requires Apps to be targeting API 31+ now, which cordova-android@10 does not support. So I'd encourage you to install cordova-android@11, which will require the following changes:

1. `JAVA_HOME` pointing to a JDK 11 install
2. Install Build tools 32.0.0 and the `PATH` containing `$ANDROID_HOME/build-tools/32.0.0` instead of `30.0.3`.
3. Remove `tools/bin` and `tools` from `PATH` variable.
4. Install `cmdline-tools` from SDK Manager, and add `$ANDROID_HOME/cmdline-tools/7.0/bin` to your `PATH` variable. The version may vary, but you can probably choose the latest version.
5. Uninstall `cordova-plugin-whitelist`, which would have been automatically installed by the Cordova CLI, a package that have recently became obsolete as of cordova-android@11.

**Comment by leonelgregioauler** [comment]
Hello !

Do you need to include this line, em Path:

![image](https://github.com/apache/cordova/assets/12154804/32775794-8a4a-4ebb-b27b-db60d8640487)
